mep3_driver/src/robot_hardware_interface.cpp

```diff
@@ -0,0 +1,139 @@
+        mOdomRightOverflow = 0;
+
+        motion_board_.init();
+        motion_board_.start();
+
+        motion_board_.set_kp_left(kp_left_);
+        motion_board_.set_ki_left(ki_left_);
+        motion_board_.set_kd_left(kd_left_);
+
+        motion_board_.set_kp_right(kp_right_);
+        motion_board_.set_ki_right(ki_right_);
+        motion_board_.set_kd_right(kd_right_);
+
+        return hardware_interface::return_type::OK;
+    }
+
+    hardware_interface::return_type RobotHardwareInterface::stop()
+    {
+        status_ = hardware_interface::status::STOPPED;
+        motion_board_.halt();
+
+        return hardware_interface::return_type::OK;
+    }
+
+    std::vector<hardware_interface::StateInterface> RobotHardwareInterface::export_state_interfaces()
+    {
+        std::vector<hardware_interface::StateInterface> interfaces;
+        interfaces.emplace_back(hardware_interface::StateInterface("left_motor", hardware_interface::HW_IF_POSITION, &mLeftWheelPositionState));
+        interfaces.emplace_back(hardware_interface::StateInterface("right_motor", hardware_interface::HW_IF_POSITION, &mRightWheelPositionState));
+        return interfaces;
+    }
+
+    std::vector<hardware_interface::CommandInterface> RobotHardwareInterface::export_command_interfaces()
+    {
+        std::vector<hardware_interface::CommandInterface> interfaces;
+        interfaces.emplace_back(hardware_interface::CommandInterface("left_motor", hardware_interface::HW_IF_VELOCITY, &mLeftWheelVelocityCommand));
+        interfaces.emplace_back(hardware_interface::CommandInterface("right_motor", hardware_interface::HW_IF_VELOCITY, &mRightWheelVelocityCommand));
+        return interfaces;
+    }
+
+    hardware_interface::return_type RobotHardwareInterface::read()
+    {
+        // Read encoder data from the robot
+        int32_t tmp_left, tmp_right;
+        std::tie(tmp_left, tmp_right) = motion_board_.get_encoders();
+
+        const int32_t leftWheelRaw = tmp_left;
+        const int32_t rightWheelRaw = tmp_right;
+
+        // Handle overflow
+        if (llabs((int64_t)mPrevLeftWheelRaw - (int64_t)leftWheelRaw) > POW2(31) - 1)
+            mOdomLeftOverflow = (mPrevLeftWheelRaw > 0 && leftWheelRaw < 0) ? mOdomLeftOverflow + 1 : mOdomLeftOverflow - 1;
+        if (llabs((int64_t)mPrevRightWheelRaw - (int64_t)rightWheelRaw) > POW2(31) - 1)
+            mOdomRightOverflow = (mPrevRightWheelRaw > 0 && rightWheelRaw < 0) ? mOdomRightOverflow + 1 : mOdomRightOverflow - 1;
+        const int64_t leftWheelCorrected = mOdomLeftOverflow * POW2(32) + leftWheelRaw;
+        const int64_t rightWheelCorrected = mOdomRightOverflow * POW2(32) + rightWheelRaw;
+        const double leftWheelRad = leftWheelCorrected / (ENCODER_RESOLUTION / (2 * M_PI));
+        const double rightWheelRad = rightWheelCorrected / (ENCODER_RESOLUTION / (2 * M_PI));
+
+        mLeftWheelPositionState = leftWheelRad;
+        mRightWheelPositionState = rightWheelRad;
```

Comment:
```suggestion
        left_wheel_position_state_ = left_wheel_rad;
        right_wheel_position_state_ = right_wheel_rad;
```
S obzirom da tek krecemo projekat imas tu cast da odaberes konvenciju, ali mora biti konzistentno :)

Reply:
Sve prebaceno na underscore.